platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventHashedData.java

```diff
@@ -342,28 +330,125 @@ public NodeId getCreatorId() {
         return creatorId;
     }
 
+    /**
+     * Get the address book round of the event.
+     *
+     * @return the address book round of the event
+     */
+    public long getAddressBookRound() {
+        return addressBookRound;
+    }
+
+    /**
+     * Get the event descriptor for the self parent.
+     *
+     * @return the event descriptor for the self parent
+     */
+    public EventDescriptor getSelfParent() {
+        return selfParent;
+    }
+
+    /**
+     * Get the event descriptors for the other parents.
+     *
+     * @return the event descriptors for the other parents
+     */
+    public List<EventDescriptor> getOtherParents() {
+        return otherParents;
+    }
+
+    /**
+     * Get the self parent generation
+     *
+     * @return the self parent generation
+     */
     public long getSelfParentGen() {
-        return selfParentGen;
+        if (selfParent == null) {
+            return -1;
+        }
+        return selfParent.getGeneration();
     }
 
+    /**
+     * Get the maximum generation of the other parents.
+     *
+     * @return the maximum generation of the other parents
+     * @deprecated this method should be replaced since there can be multiple other parents.
+     */
     public long getOtherParentGen() {
```

Comment:
It's not immediately obvious to me if we will want to have this method return the maximum other parent generation. We may not even keep this method around when we actually support multiple other parents. But I'm ok punting this design decision until later.

Reply:
Agreed.  I will throw an unsupported operation exception if there are multiple other parents so that we are forced to rewrite this method when we actually start using multiple other parents.